docs/adrs/9999-use-playwright.md

```diff
@@ -0,0 +1,82 @@
+
+The widely accepted explanation for all of the sporadic test failures (and a significant portion of the consistent failures) is that our integration test code has been written in a way that creates race conditions that are experienced when executed; it's likely that the upgraded version of the Cypress library contains a faster runtime that exposes those race conditions to a higher degree. As the code within our test suite has evolved over a long period and many contributors to it are no longer on the project, the chances of understanding _why_ it was written in the way it was seems impossible.
+
+## Considered Alternatives
+
+- Continue using existing version of Cypress
+- Upgrade to current version of Cypress
+- Convert integration tests to Playwright, gradually
+- **Convert integration tests to Playwright, all at once**
+
+## Decision Outcome
+
+First, we've decided that the current situation with using a long out-of-date version of Cypress is untenable. While it still maintains some functionality -- it has caught legitimate errors that in pull requests would have otherwise been merged into the `main` branch of the code base -- the existing level of flakiness is not something we should accept. Additionally, there are a broad range of developers on the project touching this code, and very little concept of standards or best practices; it can reasonably be assumed that the rate of sporadic and consistent test failures will increase, as will the rate of false negative tests that don't actually cover what we believe it does. Finally, being stuck on an outmoded piece of code makes us functionally incapable of any security fixes; the longer we go in this state, the more painful any forced update will eventually be.
+
+Give that, if we are going to need to re-write a significant portion (if not all) of our integration test code base, it makes sense at this juncture to evaluate other tools that may be a better fit for our application and needs. [Playwright](https://playwright.dev/) is a well-supported integration test library that's seeing rapid community adoption.
+
+Comparable to Cypress in aim, Playwright has the benefit of improved tooling around automatic test generation (increasing developer velocity), as well as much stronger, interactive reporting of errors as opposed to Cypress' video output. Unlike Cypress, the interface of code written for Playwright is similar to that of React Testing Library, which we're already using for our front end unit tests. The compile-time and runtime for Playwright tests, especially locally, is also improved over the Cypress baseline, providing a significantly shorter feedback loop that allows for better developer iteration.
```

Comment:
**thought**: Idk if we want to mention it in this adr or not, but I recently learned about the [Cypress Testing Library package](https://testing-library.com/docs/cypress-testing-library/intro/) that provides an interface that is similar to what we use for our unit tests. I don't think this alone would be a reason to stay on cypress, and in fact I'd rather switch, but it might be good to mention it as something that we could possibly use if we decided to re-write our cypress tests.